### PR TITLE
Add a page where lock creators can view keyholder metadata

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This draft PR is the current working state of the keyholder metadata page. It's close to MVP status, I plan on integrating Locksmith in a follow-up PR to keep things manageable and bite-sized.

It will display a table like this, which expands to fit various metadata, leaving blank spaces for any key that doesn't have a given field.

image

# Issues

Fixes #
Refs #5209 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Nice! Thanks a lot @cnasc !

File: unlock-app/src/components/content/KeyMetadataContent.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+interface ReduxState {
+  account: AccountType
+  network: Network
+  router: Router
+}
+
+export const mapStateToProps = ({ account, network, router }: ReduxState) => {
+  // URL formatted like: ?locks=address1,address2,address3
+  const search = queryString.parse(router.location.search, {
```
Comment (julien51) [inline unlock-app/src/components/content/KeyMetadataContent.tsx]: Hooo that's cool! I did not know we could parse URLs like this.

File: unlock-app/src/components/interface/MetadataTable.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+interface Props {
+  // The keys to the metadata object, in the order they will be displayed.
+  columns: string[]
+  metadata: KeyMetadata[]
+}
+
+/**
+ * Applied to itself, yields "Camel Case To Title"
+ */
+export function camelCaseToTitle(s: string): string {
```
Comment (julien51) [inline unlock-app/src/components/interface/MetadataTable.tsx]: Add note to move to helper/utils?

File: unlock-app/src/pages/keymetadata.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import React from 'react'
```
Comment (julien51) [inline unlock-app/src/pages/keymetadata.js]: can we rename this page as `/members`?

**Comment by cnasc** [comment]
image

Here's an actual screenshot of it running in staging

File: unlock-app/src/pages/members.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import React from 'react'
+import KeyMetadataContent from '../components/content/KeyMetadataContent'
+
+const KeyMetadata = () => <KeyMetadataContent />
```
Comment (julien51) [inline unlock-app/src/pages/members.js]: I am going to be a pain here but let;s stick to `MembersContent` please...

File: unlock-app/src/components/interface/MetadataTable.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  )
+}
+
+const Table = styled.table`
+  width: 100%;
+  border-collapse: collapse;
+`
+
+const Thead = styled.thead`
+  background-color: var(--brand);
```
Comment (julien51) [inline unlock-app/src/components/interface/MetadataTable.tsx]: Do you mind if we try to use the same layout that is used for other headers on say /dashbaord or /logs to remain consistent?

File: unlock-app/src/components/content/KeyMetadataContent.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    lock.keys.forEach((key: any) => {
+      metadata.push({
+        lockName: lock.name,
+        expiration: expirationAsDate(parseInt(key.expiration)),
+        keyholderAddress: key.owner.address,
+      })
+    })
+  })
+
+  // TODO: get additional metadata from locksmith if it exists, for each key.
```
Comment (julien51) [inline unlock-app/src/components/content/KeyMetadataContent.tsx]: Great! Thanks, lets work on this right after this one gets merged please!

File: unlock-app/src/components/interface/MetadataTable.tsx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  color: var(--white);
+`
+
+const Tbody = styled.tbody`
+  color: var(--slate);
+  & > tr:nth-child(even) {
```
Comment (julien51) [inline unlock-app/src/components/interface/MetadataTable.tsx]: Let's stay consistent here too and not have alternating colors...

**Comment by cnasc** [comment]
image

**Comment by julien51** [comment]
Perfect! Thanks!

**Pull request merged**